src/App.js

```diff
@@ -396,7 +389,13 @@ class App extends Component {
     } else {
       should_prov = false;
     }
-    this.setState({ shouldProvision: should_prov })
+
+    this.setState({
+      siteEditShow: false,
+      selectedSite: null,
+      yaml: doc,
+      shouldProvision: should_prov
+    })
```

Comment:
Hmm, this does not fix it. It seems the issue is that for some reason here we are checking if the current `should_provision` settings contains this boxID but we should actually be adding this boxID to the settings. As well as just setting the shouldProvision state to `true` since we know it should be.

Something like this:
```Javascript
this.setState({
  yaml: doc,
  homesteadSettingsShow: false,
  shouldProvision: true,
})

let shouldProvision = settings.get('should_provision')
if (shouldProvision) {
  shouldProvision.push(boxID)
} else {
  shouldProvision = [boxID]
}
settings.set('should_provision', shouldProvision)
```

Though that could probably be improved to make sure duplicate boxIDs are not added.